src/gcp_scanner/scanner.py

```diff
@@ -221,15 +222,27 @@ def crawl_loop(initial_sa_tuples: List[Tuple[str, Credentials, List[str]]],
       # Get storage buckets
       if is_set(scan_config, 'storage_buckets'):
         dump_file_names = None
+        dump_iam_file_names = None
         if scan_config is not None:
           obj = scan_config.get('storage_buckets', None)
           if obj is not None and obj.get('fetch_file_names', False) is True:
             dump_file_names = open(gcs_output_path, 'w', encoding='utf-8')
-        project_result['storage_buckets'] = crawl.get_bucket_names(project_id,
+          if obj is not None and obj.get('fetch_buckets_iam', False) is True: 
+            dump_iam_file_names = open(gcs_iam_output_path, 'w', encoding='utf-8')
+          if obj is not None and obj.get('buckets_iam', False) is True:
+            buckets_iam_policy = dict()
+            bucket_names = crawl.get_bucket_names(project_id, credentials, dump_file_names)
+            for bucket_name in bucket_names:
+              buckets_iam_policy[bucket_name] = crawl.get_bucket_iam(bucket_name, credentials, dump_iam_file_names)
+            project_result['storage_buckets']['iam_policies'] = buckets_iam_policy
+        project_result['storage_buckets']['buckets'] = crawl.get_bucket_names(project_id,
```

Comment:
You can have extra flag for get_bucket_names like (dump_iam_policies) and then if the flag is set call get_bucket_iam

Reply:
Then I have a strange question, I understand correctly that if I need to get the IAM policy (of which there will be 100% less than 50 pieces) and a list of 1 million files, then I'd better put them in 1 file than have 1 file with all the information about the project and separately a list of files nested in a bucket?

That is, parsing a file for 1000 lines (all information about the project + buckets) will be more difficult than parsing 1 million lines of file names + 50-100 lines of IAM Policy?

Reply:
And in any case, there will be two requests to the API, since the information about the files and the information about the IAM are in different endpoints.

Reply:
> And in any case, there will be two requests to the API, since the information about the files and the information about the IAM are in different endpoints. getIamPolicy() and objects()

As per current logic, we are going to have 3 requests when the user sets both `buckets_iam` and `fetch_file_names`. The scanner will query all buckets via `crawl.get_bucket_names` at line 234 and at line 238. This might be a significant performance issue if we have a lot of buckets to query.